util/guard/guard.go

```diff
@@ -0,0 +1,20 @@
+package guard
+
+import (
+	"runtime/debug"
+)
+
+// Minimal logger contract; avoids depending on any specific logging package.
+type Logger interface{ Errorf(string, ...any) }
+
+// Run executes fn and recovers a panic, logging a component-specific message.
+func Run(fn func(), log Logger, msg string) {
```

Comment:
Maybe just straight-up call it `RecoverAndLog`? Helps avoid a click in to fully understand what it does.